code/datums/components/infective.dm

```diff
@@ -8,43 +8,78 @@
 	var/weak_infection_chance = 10
 
 
-/datum/component/infective/Initialize(list/datum/disease/_diseases, expire_in, weak = FALSE)
-	if(islist(_diseases))
-		diseases = _diseases
-	else
-		diseases = list(_diseases)
+/datum/component/infective/Initialize(list/datum/disease/diseases, expire_in, weak = FALSE, weak_infection_chance = 10)
+	if(!ismovable(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	if(!islist(diseases))
+		diseases = islist(diseases)
```

Comment:
```suggestion
	if(!islist(diseases))
		diseases = FALSE
```
?

Reply:
I don't know what you're suggesting but it won't work. The `diseases` var has to be a list, hence the name. This couple lines are just here to allow contributors to add more instances of the infective component in the game without having to wrap a single disease inside a list during the `AddComponent()` call.

Reply:
Ah, I was confused because this line could only ever resolve to FALSE, since !islist already exclused anything else